tests/model_fields/test_foreignkey.py

```diff
@@ -128,3 +128,37 @@ class Bar(models.Model):
 
         with self.assertRaisesMessage(ValueError, 'Cannot resolve output_field'):
             Foo._meta.get_field('bar').get_col('alias')
+
+    @isolate_apps('model_fields')
+    def test_non_local_model_relation(self):
+        class BaseModel(models.Model):
+            key = models.IntegerField(unique=True)
+
+        class ChildModel(BaseModel):
+            pass
+
+        class RelatedModel(models.Model):
+            ref = models.ForeignKey(ChildModel, on_delete=models.CASCADE, to_field='key')
+
+        with self.assertRaisesMessage(
+            ValueError,
+            "ForeignKey 'model_fields.RelatedModel.ref' pointing to a "
+            "non local remote field 'model_fields.BaseModel.key'.",
+        ):
+            RelatedModel._meta.get_field('ref').resolve_related_fields()
+
+    @isolate_apps('model_fields')
+    def test_local_model_relation(self):
+        class BaseModel(models.Model):
+            key = models.IntegerField(unique=True)
+
+            class Meta:
+                abstract = True
+
+        class ChildModel(BaseModel):
+            pass
+
+        class RelatedModel(models.Model):
+            ref = models.ForeignKey(ChildModel, on_delete=models.CASCADE, to_field='key')
+
+        self.assertIsNotNone(RelatedModel._meta.get_field('ref').resolve_related_fields())
```

Comment:
Not sure this test adds anything over the ones we already have for `to_field` and abstract model field inheritance.